drivers/network/dd/e1000/debug.c

```diff
@@ -0,0 +1,167 @@
+        RETURN_X(OID_GEN_DRIVER_VERSION);
+        RETURN_X(OID_GEN_MAXIMUM_TOTAL_SIZE);
+        RETURN_X(OID_GEN_PROTOCOL_OPTIONS);
+        RETURN_X(OID_GEN_MAC_OPTIONS);
+        RETURN_X(OID_GEN_MEDIA_CONNECT_STATUS);
+        RETURN_X(OID_GEN_MAXIMUM_SEND_PACKETS);
+        RETURN_X(OID_GEN_VENDOR_DRIVER_VERSION);
+        RETURN_X(OID_GEN_SUPPORTED_GUIDS);
+        RETURN_X(OID_GEN_NETWORK_LAYER_ADDRESSES);
+        RETURN_X(OID_GEN_TRANSPORT_HEADER_OFFSET);
+        RETURN_X(OID_GEN_MACHINE_NAME);
+        RETURN_X(OID_GEN_RNDIS_CONFIG_PARAMETER);
+        RETURN_X(OID_GEN_VLAN_ID);
+
+        /* Optional OIDs */
+        RETURN_X(OID_GEN_MEDIA_CAPABILITIES);
+        RETURN_X(OID_GEN_PHYSICAL_MEDIUM);
+
+        /* Required statistics OIDs */
+        RETURN_X(OID_GEN_XMIT_OK);
+        RETURN_X(OID_GEN_RCV_OK);
+        RETURN_X(OID_GEN_XMIT_ERROR);
+        RETURN_X(OID_GEN_RCV_ERROR);
+        RETURN_X(OID_GEN_RCV_NO_BUFFER);
+
+        /* Optional statistics OIDs */
+        RETURN_X(OID_GEN_DIRECTED_BYTES_XMIT);
+        RETURN_X(OID_GEN_DIRECTED_FRAMES_XMIT);
+        RETURN_X(OID_GEN_MULTICAST_BYTES_XMIT);
+        RETURN_X(OID_GEN_MULTICAST_FRAMES_XMIT);
+        RETURN_X(OID_GEN_BROADCAST_BYTES_XMIT);
+        RETURN_X(OID_GEN_BROADCAST_FRAMES_XMIT);
+        RETURN_X(OID_GEN_DIRECTED_BYTES_RCV);
+        RETURN_X(OID_GEN_DIRECTED_FRAMES_RCV);
+        RETURN_X(OID_GEN_MULTICAST_BYTES_RCV);
+        RETURN_X(OID_GEN_MULTICAST_FRAMES_RCV);
+        RETURN_X(OID_GEN_BROADCAST_BYTES_RCV);
+        RETURN_X(OID_GEN_BROADCAST_FRAMES_RCV);
+        RETURN_X(OID_GEN_RCV_CRC_ERROR);
+        RETURN_X(OID_GEN_TRANSMIT_QUEUE_LENGTH);
+        RETURN_X(OID_GEN_GET_TIME_CAPS);
+        RETURN_X(OID_GEN_GET_NETCARD_TIME);
+        RETURN_X(OID_GEN_NETCARD_LOAD);
+        RETURN_X(OID_GEN_DEVICE_PROFILE);
+        RETURN_X(OID_GEN_INIT_TIME_MS);
+        RETURN_X(OID_GEN_RESET_COUNTS);
+        RETURN_X(OID_GEN_MEDIA_SENSE_COUNTS);
+        RETURN_X(OID_GEN_FRIENDLY_NAME);
+        RETURN_X(OID_GEN_MINIPORT_INFO);
+        RETURN_X(OID_GEN_RESET_VERIFY_PARAMETERS);
+
+        /* IEEE 802.3 (Ethernet) OIDs */
+        //RETURN_X(NDIS_802_3_MAC_OPTION_PRIORITY);     /*Duplicate ID */
+
+        RETURN_X(OID_802_3_PERMANENT_ADDRESS);
+        RETURN_X(OID_802_3_CURRENT_ADDRESS);
+        RETURN_X(OID_802_3_MULTICAST_LIST);
+        RETURN_X(OID_802_3_MAXIMUM_LIST_SIZE);
+        RETURN_X(OID_802_3_MAC_OPTIONS);
+        RETURN_X(OID_802_3_RCV_ERROR_ALIGNMENT);
+        RETURN_X(OID_802_3_XMIT_ONE_COLLISION);
+        RETURN_X(OID_802_3_XMIT_MORE_COLLISIONS);
+        RETURN_X(OID_802_3_XMIT_DEFERRED);
+        RETURN_X(OID_802_3_XMIT_MAX_COLLISIONS);
+        RETURN_X(OID_802_3_RCV_OVERRUN);
+        RETURN_X(OID_802_3_XMIT_UNDERRUN);
+        RETURN_X(OID_802_3_XMIT_HEARTBEAT_FAILURE);
+        RETURN_X(OID_802_3_XMIT_TIMES_CRS_LOST);
+        RETURN_X(OID_802_3_XMIT_LATE_COLLISIONS);
+
+        /* IEEE 802.11 (WLAN) OIDs */
```

Comment:
WLAN OIDs in an Ethernet driver? :open_mouth:

Reply:
To log something you should know the name for it,
and if we are going to leave names out because we 'dont expect them here', then how will we know if the unexpected happens?